python/ray/air/_internal/accelerator_utils/npu.py

```diff
@@ -0,0 +1,84 @@
+import os
+from typing import List
+
+import torch
+import ray
+from ray._private.accelerators.npu import (
+    ASCEND_RT_VISIBLE_DEVICES_ENV_VAR,
+    NPU_TORCH_PACKAGE_AVAILABLE,
+)
+from ray.air._internal.accelerator_utils.device_manager import TorchDeviceManager
+
+if NPU_TORCH_PACKAGE_AVAILABLE:
+    import torch_npu  # noqa: F401
+
+
+class NPUTorchDeviceManager(TorchDeviceManager):
+    """Ascend NPU device manager"""
+
+    @staticmethod
+    def get_accelerator_name() -> str:
+        return "NPU"
+
+    @staticmethod
+    def get_device_type() -> str:
+        return "npu"
+
+    @staticmethod
+    def get_device_module():
+        if not NPU_TORCH_PACKAGE_AVAILABLE:
+            raise ModuleNotFoundError(
+                "torch-npu is not installed so that npu is not available, "
+                "please use `pip install torch-npu` to install torch-npu."
+            )
+        return torch.npu
+
+    @staticmethod
+    def is_device_available() -> bool:
+        if not NPU_TORCH_PACKAGE_AVAILABLE:
+            return False
+
+        return torch.npu.is_available()
+
+    @staticmethod
+    def get_devices() -> List[torch.device]:
```

Comment:
There seems to have duplicated code in `get_device` method for GPU and NPU. Can we reuse some of them?

Reply:
NPU and GPU appear to be similar, but HPU is quite different. So we did not abstract out a common parent class, as this makes it more convenient for device integration.